### PR TITLE
feat(state): implement hero state + refactoring/update

### PR DESCRIPTION
**Pull request by stephenlautier**

### CODE REVIEW

File: Heroes.Server/Heroes.Server.csproj [hunk 1]
```diff
@@ -21,7 +21,9 @@
     <PackageReference Include="Grace.AspNetCore.Hosting" Version="7.0.1" />
     <PackageReference Include="Grace.DependencyInjection.Extensions" Version="7.0.1" />
     <PackageReference Include="graphiql" Version="1.2.0" />
-    <PackageReference Include="GraphQL" Version="2.4.0" />
+    <PackageReference Include="GraphQL" Version="3.0.0-preview-1552" />
+    <PackageReference Include="GraphQL.Server.Transports.AspNetCore.NewtonsoftJson" Version="3.5.0-alpha0060" />
+    <PackageReference Include="GraphQL.Server.Ui.Playground" Version="3.5.0-alpha0060" />
```
Comment (claylaut) [inline Heroes.Server/Heroes.Server.csproj:26]: Alpha/previews :D

File: heroes.client.ngx-app/src/app/shared/state/hero.service.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import * as _ from "lodash";
+import gql from "graphql-tag";
+import { Observable, of } from "rxjs";
+import { Injectable } from "@angular/core";
+
+import { HeroRoleType, Hero } from "./hero.model";
+import { AppApolloClient } from "../gql/apollo.client";
+import { ApolloBase } from "apollo-angular";
```
Comment (claylaut) [inline heroes.client.ngx-app/src/app/shared/state/hero.service.ts]: sort

File: heroes.client.ngx-app/src/app/shared/state/hero.service.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import * as _ from "lodash";
+import gql from "graphql-tag";
+import { Observable, of } from "rxjs";
+import { Injectable } from "@angular/core";
+
+import { HeroRoleType, Hero } from "./hero.model";
+import { AppApolloClient } from "../gql/apollo.client";
```
Comment (claylaut) [inline heroes.client.ngx-app/src/app/shared/state/hero.service.ts:7]: sort

File: heroes.client.ngx-app/src/app/shared/state/hero.service.ts (hunk 2 above)
Comment (stephenlautier) [reply heroes.client.ngx-app/src/app/shared/state/hero.service.ts]: done

**Review by claylaut — DISMISSED** [review]

**Review by claylaut — DISMISSED** [review]

**Pull request merged**